File: src/main/java/fi/finlit/edith/ui/services/NoteAdditionFailedException.java
/*
 * Copyright (c) 2009 Mysema Ltd.
 * All rights reserved.
 * 
 */
package fi.finlit.edith.ui.services;

import fi.finlit.edith.domain.SelectedText;

/**
 * NoteAdditionFailedException provides
 *
 * @author tiwe
 * @version $Id$
 */
@SuppressWarnings("serial")
public class NoteAdditionFailedException extends Exception{
    
    private final SelectedText selectedText;
    
    private final String localId;
    
    public NoteAdditionFailedException(SelectedText selectedText, String localId, boolean startMatched, boolean endMatched) {
        super(createDescription(localId, startMatched, endMatched));
        this.selectedText = selectedText;
        this.localId = localId;
    }
    
    private static String createDescription(String localId, boolean startMatched, boolean endMatched){
        StringBuilder builder = new StringBuilder();
        builder.append("Failed to add selected text for note #").append(localId);
        if (!startMatched){
            builder.append("; start was not matched");
        }
        if (!endMatched){
            builder.append("; end was not matched");
        }
        return builder.toString();
    }

    public SelectedText getSelectedText() {
        return selectedText;
    }

    public String getLocalId() {
        return localId;
    }
    
}

File: src/main/scala/fi/finlit/edith/ui/services/NoteAdditionFailedException.scala
package fi.finlit.edith.ui.services

import fi.finlit.edith.dto.SelectedText
import NoteAdditionFailedException._
import scala.reflect.BeanProperty
//remove if not needed
import scala.collection.JavaConversions._

object NoteAdditionFailedException {

  private def createDescription(localId: String, startMatched: Boolean, endMatched: Boolean): String = {
    var builder = new StringBuilder()
    builder.append("Failed to add selected text for note #").append(localId)
    if (!startMatched) {
      builder.append("; start was not matched")
    }
    if (!endMatched) {
      builder.append("; end was not matched")
    }
    builder.toString
  }
}

class NoteAdditionFailedException(@BeanProperty val selectedText: SelectedText, 
    @BeanProperty val localId: String, 
    startMatched: Boolean, 
    endMatched: Boolean) extends Exception(createDescription(localId, startMatched, endMatched)) {


}
